core/tests/webdriverio_utils/StoryEditorPage.js

```diff
@@ -423,6 +432,59 @@ var StoryEditorPage = function() {
       'Node Description Input', nodeDescriptionInputField, nodeDescription);
   };
 
+  this.setNodePlannedPublicationDate = async function(dateString) {
+    await waitFor.visibilityOf(
+      plannedPublicationDateInput,
+      'PlannedPublicationDateInput takes too long to be visible'
+    );
+
+    await action.setValue(
+      'Planned Publication Date Input', plannedPublicationDateInput,
+      dateString);
+  };
+
+  this.setNodeStatusToReadyToPublish = async function() {
+    await waitFor.visibilityOf(
+      markAsReadyToPublishButton,
+      'Mark As Ready To Publish button taking too long to appear');
+    await action.click(
+      'Mark As Ready To Publish Button', markAsReadyToPublishButton);
+  };
+
+  this.setNodeStatusToDraft = async function() {
+    await waitFor.visibilityOf(
+      markAsDraftButton, 'Mark As Draft button taking too long to appear');
+    await action.click('Mark As Draft Button', markAsDraftButton);
+  };
+
+  this.publishNodeChanges = async function() {
+    await waitFor.visibilityOf(
+      publishChangesButton,
+      'Publish Changes button taking too long to appear');
+    await action.click('Publish Changes Button', publishChangesButton);
+  };
+
+  this.publishNodes = async function(chapterIndex) {
+    await action.select(
+      'Publish Upto Chapter Dropdown',
+      publishUptoChaptersDropdownSelector, Number(chapterIndex) + 1);
+    await waitFor.visibilityOf(
+      publishChapterButton,
+      'Publish Chapters Button taking too long to appear');
+    await action.click('Publish Chapters Button', publishChapterButton);
```

Comment:
I think `action.click` takes care of `waitFor.visibilityOf(publishChapterButton, ...)`, can you please confirm and remove the above visibility check as needed?

Reply:
Done